primaryHandler.go

```diff
@@ -563,27 +563,34 @@ func validateDeviceID() alice.Chain {
 	})
 }
 
-func ValidateWRP() func(http.Handler) http.Handler {
+func ValidateWRP(logger *zap.Logger) func(http.Handler) http.Handler {
 	return func(delegate http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 
 			if msg, ok := wrpcontext.GetMessage(r.Context()); ok {
-				validators := wrp.SpecValidators()
 				var err error
-				for _, v := range validators {
-					err = multierr.Append(err, v.Validate(*msg))
-				}
```

Comment:
i say keep the for loop pattern you had and add `assert.ErrorIs`
it'll help manage and make clear which error we'll like to return 400s for

Reply:
is there a different assert package? i thought assert was just for testing.

Reply:
nevermind i think i found the function i can use.

Reply:
oh sorry, I meant to say `errors.Is` and not `assert.ErrorIs`